data/mods/gen4/moves.ts

```diff
@@ -827,10 +827,13 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 	},
 	knockoff: {
 		inherit: true,
-		onAfterHit(target, source) {
-			const item = target.takeItem();
-			if (item) {
+		onAfterHit(target, source, move) {
+			if (!target.item || target.itemState.knockedOff) return;
+			const item = target.getItem();
+			if (this.singleEvent('TakeItem', item, target.itemState, target, target, move, item)) {
+				target.itemState.knockedOff = true;
 				this.add('-enditem', target, item.name, '[from] move: Knock Off', '[of] ' + source);
+				this.hint("Prior to Gen 5, Knock Off did not actually remove the target's item. Instead, the item would be unusable for the rest of the battle, and the target would not be able to obtain a new item.", true);
```

Comment:
This is a fairly long hint for our standards, especially on something that will happen after any time Knock Off itself is used. I'd rather move the hint to a situation where unusual behavior occurs.

As for a short version:
In Gens 3-4, Knock Off only makes the target's item unusable; it cannot obtain a new item.

Reply:
> especially on something that will happen after any time Knock Off itself is used

I made it so that the hint only shows the first time knock off is used in a battle. I guess the hint could be moved somewhere else but I don't know where that would be.

Reply:
Oops, missed the ``true`` argument haha